subprojects/testfx-core/src/main/java/org/testfx/util/WaitForAsyncUtils.java

```diff
@@ -367,11 +376,14 @@ public static void waitForFxEvents() {
      *
      * @param attemptsCount the number of attempts to try
      */
-    public static void waitForFxEvents(int attemptsCount) {
-        for (int attempt = 0; attempt < attemptsCount; attempt++) {
-            blockFxThreadWithSemaphore();
-            sleep(SEMAPHORE_SLEEP_IN_MILLIS, MILLISECONDS);
+    public static void waitForFxEvents(int attemptsCount, int pulses) {
+        if (Platform.isFxApplicationThread()) {
+            throw new RuntimeException("Waiting for FxEvents on the Fx-Thread is just not possible. " +
+                    "Call the root on the test Thread!");
```

Comment:
What is "root" referring to here? Can we make it a bit more clear? Just in case anyone encounters this error in trying to use TestFX, I like to be very explicit (erring on the side of pedantic).

Reply:
That's not pedantic, this message is meant to help the user (and us) to find the issue as fast as possible. So it shouldn't be confusing.  
The root is the method call, that actuallty triggered `waitForFxEvents`. The point is, that the method might be anywhere in the call tree:
- direct call of the user in the test case (e.g. via `interact(()->moveTo(...))` or `Platform.runLater('any RobotMethod')` the `Trace of caller`-part of the exception should point to the line
- indirect call in the TestFx framework (-> bug)
- direct call in the TestFx framework itself (-> bug)
The latter two cases might be a bit tricky to identify, but shouldn't occur that frequently (if we have done it right :-) ).

```java
throw new RuntimeException("Waiting for FxEvents on the Fx-Thread is just not possible. " +
                    "See stacktrace below for the caller (most probably 'Trace of caller')." +
                    "Try calling this method on the test thread (or any background thread)!");
``` 
This message would point to and user error first. Any better suggestions? (There are still some check-ins left, so take your time :-) )

Reply:
My wording would probably be `Waiting for events on the "JavaFX Application Thread" is not possible (nor advisable). Instead, call "waitForFxEvents" on a test thread (or any other background thread). See stacktrace(s) below to find the bad call to "waitForFxEvents".`

But yours is totally acceptable :).